src/SurfaceFluxes.jl

```diff
@@ -903,8 +930,95 @@ function recover_profile(
     num2 = -UF.psi(uf, Z / L_MO, transport)
     num3 = UF.psi(uf, z0(sc, transport) / L_MO, transport)
     Σnum = num1 + num2 + num3
-    ΔX = X_in - X_sfc
-    return Σnum * compute_physical_scale_coeff(param_set, sc, L_MO, transport, uft, scheme) * _π_group⁻¹ * ΔX + X_sfc
+    return Σnum * compute_physical_scale_coeff(param_set, sc, L_MO, transport, uft, scheme) / von_karman_const *
+           (X_in - X_sfc) + X_sfc
+end
+
+
+"""
+    recover_profile(param_set, sc, ca, L_MO, Z, X_in, X_sfc, transport, uft, scheme) 
+
+Recover profiles of variable X given values of Z coordinates, as long as Z > d. 
+Follows Nishizawa equation (21, 22) 
+Canopy modification follows Physick and Garratt (1995) equation (8, 9)
+## Arguments
+  - param_set: Abstract Parameter Set containing physical, thermodynamic parameters.
+  - sc: Container for surface conditions based on known combination
+        of the state vector, and {fluxes, friction velocity, exchange coefficients} for a given experiment
+  - L_MO: Monin-Obukhov length
+  - ca: Canopy Type (e.g. NoCanopy, SparseCanopy, DenseCanopy)
+  - Z: Z coordinate(s) (within surface layer) for which variable values are required
+  - X_in, X_sfc: For variable X, values at interior and surface nodes
+  - transport: Transport type, (e.g. Momentum or Heat, used to determine physical scale coefficients)
+  - uft: A Universal Function type, (returned by, e.g., Businger())
+  - scheme: Discretization scheme (currently supports FD and FV)
+
+# TODO: add tests
+# TODO: Verify that all current RSL models fall into this general code pattern, then 𝜙 can be abstracted
+"""
+function recover_profile(
+    param_set::APS,
+    sc::AbstractSurfaceConditions,
+    ca::Union{SparseCanopy, DenseCanopy},
+    L_MO::FT,
+    Z,
+    X_in,
+    X_sfc,
+    transport,
+    uft::UF.AUFT,
+    scheme::Union{FVScheme, FDScheme},
+) where {FT}
+    z_star = ca.z_star
+    d = ca.d
+    @assert isless.(Z - d, z_in(sc))
```

Comment:
This assertion looks incorrect-- any broadcasted object is not going to return a Bool

Reply:
Maybe `all()` is missing?